src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -0,0 +1,330 @@
+        break;
+
+    case Commands::AddMoreTime::Id:
+        ChipLogDetail(Zcl, "MicrowaveOvenControl:  Entering AddMoreTime");
+
+        CommandHandlerInterface::HandleCommand<Commands::AddMoreTime::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleAddMoreTime(ctx, req); });
+        break;
+    }
+}
+
+void Instance::HandleSetCookingParameters(HandlerContext & ctx, const Commands::SetCookingParameters::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "MicrowaveOvenControl:  HandleSetCookingParameters");
+    Status status;
+    uint8_t opState;
+    auto & CookMode     = req.cookMode;
+    auto & CookTime     = req.cookTime;
+    auto & PowerSetting = req.powerSetting;
+
+    // TODO(#30609): check if the input cook mode value is invalid
+
+    if (CookTime.HasValue() && (!IsCookTimeInRange(CookTime.Value())))
+    {
+        status = Status::InvalidCommand;
+        ChipLogError(Zcl, "Failed to set cookTime, cookTime value is out of range");
+        goto exit;
+    }
+
+    if (PowerSetting.HasValue() &&
+        (!IsPowerSettingInRange(PowerSetting.Value(), mDelegate->GetMinPower(), mDelegate->GetMaxPower())))
+    {
+        status = Status::InvalidCommand;
+        ChipLogError(Zcl, "Failed to set cookPower, cookPower value is out of range");
+        goto exit;
+    }
+
+    // Get Operational State instance
+    gOperationalStateInstance = GetOPInstance(mEndpointId);
+    if (gOperationalStateInstance == nullptr)
+    {
+        status = Status::InvalidInState;
+        ChipLogError(Zcl, "Microwave Oven Control: Operational State instance does not exist");
+        goto exit;
+    }
+
+    opState = gOperationalStateInstance->GetCurrentOperationalState();
+    if (opState == to_underlying(OperationalStateEnum::kStopped))
+    {
+        uint8_t reqCookMode     = 0;
+        uint32_t reqCookTime    = 0;
+        uint8_t reqPowerSetting = 0;
+        if (CookMode.HasValue())
+        {
+            // TODO: set Microwave Oven cooking mode by CookMode.Value().
+        }
+        else
+        {
+            // TODO: set Microwave Oven cooking mode to normal mode.
+        }
+
+        if (CookTime.HasValue())
+        {
+            reqCookTime = CookTime.Value();
+        }
+        else
+        {
+            // set Microwave Oven cooking time to 30 seconds(default).
+            reqCookTime = MicrowaveOvenControl::kDefaultCookTime;
+        }
+
+        if (PowerSetting.HasValue())
+        {
+            reqPowerSetting = PowerSetting.Value();
+        }
+        else
+        {
+            // set Microwave Oven cooking power to max power(default).
+            reqPowerSetting = mDelegate->GetMaxPower();
+        }
+        status = mDelegate->HandleSetCookingParametersCallback(reqCookMode, reqCookTime, reqPowerSetting);
+        goto exit;
+    }
+    else
+    {
+        status = Status::InvalidInState;
+        goto exit;
```

Comment:
I there a reason not to reverse the opstate test and outdent all the non-error code?